### PR TITLE
4KB RAM SSL support

### PR DESCRIPTION
**Pull request by anhashia**
4KB SSL incoming limit support on IoT Hub server side certificate renewal

 

# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 
  - [ ] I have merged the latest `master` branch prior to submission and re-merged as needed after I took any feedback.
  - [ ] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -109,6 +109,7 @@ The minimum requirements are for the device platform to support the following:
 - **Support SHA-256** (optional): necessary to generate the secure token for authenticating the device with the service. Different authentication methods are available and not all require SHA-256.
 - **Have a Real Time Clock or implement code to connect to an NTP server**: necessary for both establishing the TLS connection and generating the secure token for authentication.
 - **Having at least 64KB of RAM**: the memory footprint of the SDK depends on the SDK and protocol used as well as the platform targeted. The smallest footprint is achieved targeting microcontrollers.
+- **Having at least 4KB set for incoming SSL max content length buffer**: For some TLS libraries, this may be a configurable option and default may have been set as **4KB** for low memory footprint devices. During TLS handshake, IoT Hub service will send Server Hello which includes IoT Hub server side certificates. During **renewal** of IoT Hub server side certificates, check will be made to prevent this exceeding 4KB limit so that existing devices which are set for 4KB limit continue to work as before after renewal.
```
Comment (massand) [inline readme.md]: @anhashia  - I assume you meant 4KB of RAM. Please explicitly add that.
@jebrando - please take a look at the readme contents here in general for anything else we need to add in this iteration.

Comment (anhashia) [reply readme.md]: Added RAM in the text

**Comment by yphuangms** [comment]
It looks like "4KB" is too rough, if we set the SSL certificate buffer to 4096 bytes, there are chances that IoT Hub certificates total size breaks this limitation. There are several IoT Hub has 4102 bytes of total 3 certificates during handshake of TLS.
This size is critical to small RAM MCU type of device.
So could you give a specific number of bytes instead of "4KB"?

Following is the tool we used to check SSL certificates expected from target IoTHub:

`# openssl s_client -showcerts -connect xxxmyiothubxxx.azure-devices.net:8883`

CONNECTED(00000003)
depth=2 C = IE, O = Baltimore, OU = CyberTrust, CN = Baltimore CyberTrust Root
verify return:1
depth=1 C = US, ST = Washington, L = Redmond, O = Microsoft Corporation, OU = Microsoft IT, CN = Microsoft IT TLS CA 1
verify return:1
depth=0 CN = *.azure-devices.net
verify return:1
---
SSL handshake has read 4102 bytes and written 515 bytes
---

**Pull request merged**